### PR TITLE
[dfyn-lp] dfyn-lp strategy added

### PR DESCRIPTION
**Pull request by vatsalgupta13**
Fixes # .

Changes proposed in this pull request:
- dfyn-lp strategy added

### CODE REVIEW

File: src/strategies/dfyn-lp/index.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import { getAddress } from '@ethersproject/address';
+import { subgraphRequest } from '../../utils';
+
+const DFYN_SUBGRAPH_URL = {
+  '137': 'https://api.thegraph.com/subgraphs/name/ss-sonic/dfyn-v5'
+};
+
+export const author = 'vfatouros';
```
Comment (ChaituVR) [inline src/strategies/dfyn-lp/index.ts:8]:
```suggestion
export const author = 'vatsalgupta13';
```

**Review by ChaituVR — CHANGES_REQUESTED** [review]
@vatsalgupta13 I see this is built on top of `infinityprotocol-liquidity-pools`  strategy, is it possible to modify `infinityprotocol-liquidity-pools`  strategy itself with `if` condition for `scoreMultiplier` and to pass subgraph url from options, which can reduce duplicate code,
image

File: src/strategies/dfyn-lp/index.ts (hunk 1 above)
Comment (vatsalgupta13) [reply src/strategies/dfyn-lp/index.ts:8]: So basically its actually based on staked-uniswap strategy. But they hardcoded the subgraph url so I had to change the code.

Comment (ChaituVR) [reply src/strategies/dfyn-lp/index.ts:8]: > So basically its actually based on staked-uniswap strategy. But they hardcoded the subgraph url so I had to change the code.

Ya I can understand that hardcoding is not good, but i see this is minor change so we can always change the strategy and pass optional params as per our requirement 

example:
```
await subgraphRequest(	
    options.subGraphURL ? options.subGraphURL : INFINITYPROTOCOL_SUBGRAPH_URL[network],	
    params
```

```
if(options.scoreMultiplier) {
  score[userAddress] = score[userAddress] * options.scoreMultiplier
}
```
and also would be great if we can document these options inside README.md

Comment (vatsalgupta13) [reply src/strategies/dfyn-lp/index.ts:8]: Sure we can do that. Should I make the changes in staked-uniswap strategy or infinityprotocol-liquidity-pools strategy, both of them are actually the same with different subgraph requests.

Comment (ChaituVR) [reply src/strategies/dfyn-lp/index.ts:8]: Any strategy should be fine as long as the previous strategy works well and your changes are small, optional, and well documented (doesn't disturb existing voting power calculation) make sure to add some test addresses into examples.json that returns a positive score with your changes

Comment (vatsalgupta13) [reply src/strategies/dfyn-lp/index.ts:8]: Sure. Will do that.

**Comment by vatsalgupta13** [comment]
Made changes to the infinityprotocol-liquidity-pools strategy. Closing this PR.

**Pull request closed without merging**